_data/reports/2022-03-biconomy.md

```diff
@@ -0,0 +1,2632 @@
+
+[LiquidityPool.sol#L151](https://github.com/code-423n4/2022-03-biconomy/blob/db8a1fdddd02e8cc209a4c73ffbb3de210e4a81a/contracts/hyphen/LiquidityPool.sol#L151)<br>
+
+The `depositErc20` function allows setting `tokenAddress = NATIVE` and does not throw an error.<br>
+No matter the `amount` chosen, the `SafeERC20Upgradeable.safeTransferFrom(IERC20Upgradeable(tokenAddress), sender, address(this), amount);` call will not revert because it performs a low-level call to `NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`, which is an EOA, and the low-level calls to EOAs always succeed.<br>
+Because the `safe*` version is used, the EOA not returning any data does not revert either.<br>
+
+This allows an attacker to deposit infinite native tokens by not paying anything.<br>
+The contract will emit the same `Deposit` event as a real `depositNative` call and the attacker receives the native funds on the other chain.
+
+### Recommended Mitigation Steps
+
+Check `tokenAddress != NATIVE` in `depositErc20`.
+
+**[ankurdubey521 (Biconomy) confirmed and commented](https://github.com/code-423n4/2022-03-biconomy-findings/issues/55):**
+ > [HP-25: C4 Audit Fixes, Dynamic Fee Changes bcnmy/hyphen-contract#42](https://github.com/bcnmy/hyphen-contract/pull/42)
+
+**[pauliax (judge) commented](https://github.com/code-423n4/2022-03-biconomy-findings/issues/55#issuecomment-1094973634):**
+ > Great find, definitely deserves a severity of high.
+
+
+
+***
+
+## [[H-02] `LiquidityProviders.sol` The share price of the LP can be manipulated and making future liquidityProviders unable to `removeLiquidity()`](https://github.com/code-423n4/2022-03-biconomy-findings/issues/139)
+_Submitted by WatchPug_
+
+[LiquidityProviders.sol#L345-L362](https://github.com/code-423n4/2022-03-biconomy/blob/04751283f85c9fc94fb644ff2b489ec339cd9ffc/contracts/hyphen/LiquidityProviders.sol#L345-L362)<br>
+
+```solidity
+function removeLiquidity(uint256 _nftId, uint256 _amount)
+    external
+    nonReentrant
+    onlyValidLpToken(_nftId, _msgSender())
+    whenNotPaused
+{
+    (address _tokenAddress, uint256 nftSuppliedLiquidity, uint256 totalNFTShares) = lpToken.tokenMetadata(_nftId);
+    require(_isSupportedToken(_tokenAddress), "ERR__TOKEN_NOT_SUPPORTED");
+
+    require(_amount != 0, "ERR__INVALID_AMOUNT");
+    require(nftSuppliedLiquidity >= _amount, "ERR__INSUFFICIENT_LIQUIDITY");
+    whiteListPeriodManager.beforeLiquidityRemoval(_msgSender(), _tokenAddress, _amount);
+    // Claculate how much shares represent input amount
+    uint256 lpSharesForInputAmount = _amount * getTokenPriceInLPShares(_tokenAddress);
+
+    // Calculate rewards accumulated
+    uint256 eligibleLiquidity = sharesToTokenAmount(totalNFTShares, _tokenAddress);
+```
+
+[LiquidityProviders.sol#L192-L194](https://github.com/code-423n4/2022-03-biconomy/blob/db8a1fdddd02e8cc209a4c73ffbb3de210e4a81a/contracts/hyphen/LiquidityProviders.sol#L192-L194)<br>
+
+```solidity=192
```

Comment:
This is one of the lines causing a syntax error.

```suggestion
```solidity
```

Reply:
Thanks, @joshuashort!!